plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3136,6 +3136,44 @@ public void testOptimizeForPartitionedTable()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @DataProvider
+    public Object[][] tableFormatVersion()
+    {
+        return new Object[][] {
+                {1},
+                {2},
+        };
+    }
+
+    @Test
+    public void testOptimizeTableAfterDeleteWithFormatVersion2()
+    {
+        String tableName = "test_optimize_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (key integer, value varchar) WITH (format_version = 2)");
+
+        // DistributedQueryRunner sets node-scheduler.include-coordinator by default, so include coordinator
+        int workerCount = getQueryRunner().getNodeCount();
+
+        assertUpdate("INSERT INTO " + tableName + " VALUES (11, 'eleven'), (12, 'zwölf')", 2);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (13, 'trzynaście')", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (14, 'quatorze')", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (15, 'пʼятнадцять')", 1);
+
+        List<String> initialFiles = getActiveFiles(tableName);
+        assertThat(initialFiles)
+                .hasSize(4)
+                .hasSizeGreaterThan(workerCount);
+
+        assertUpdate("DELETE FROM " + tableName + " WHERE key = 11", 1);
```

Comment:
This may remove a whole file https://github.com/trinodb/trino/pull/12197 
while we want to test position deletes.

1. populate the table using `nation`
2. delete one row by `nationkey = 7`